themes/classic/_dev/js/cart.js

```diff
@@ -168,6 +168,49 @@ $(document).ready(() => {
     });
   };
 
+  //Hide 'have a code promo' text if displayed when 'close' is
+  let isPresent = $('.promo-code-button').length,
+      nameLength = 0;
+  
+  $('.cancel-promo').hide();
+
+  function labelCodeDisplayed() {
+    $('.cancel-promo').on('click', (e) => {
+      $(e.currentTarget).hide();
+      $('.promo-code-button:not(.cancel-promo)').show();
+    });
+
+    $('.promo-code-button:not(.cancel-promo)').on('click', (e) => {
+        $(e.currentTarget).hide();
+        $('.cancel-promo').show();
+    });
+  }
+
+  function isBlockPromoPresent() {
+    if ($('.block-promo-container').length != 0 && isPresent === 2) {
+      $('.cancel-promo').show();
+      $('.promo-code-button').not('.cancel-promo').hide();
+    }
+  }
+
+  setInterval( function () {
```

Comment:
Okay, thank you.
It was when you have a promo code and add it, "have a promo code" and "close" were displayed at the same time. So I add this to manage it.
The promo code is displayed from a condition in tpl file.

Reply:
Yes but you could it with the onReady event, no need for a setInterval

Reply:
I still don't understand the purpose of `setInterval` ?

Reply:
I put a setInterval because when you have a promo code and add it, both texts "have a promo code" and "closed" were displayed. Normally, only "have a promo code." is supposed to be displayed once you add your promo code.
'setInterval' was there to call again my function to fix that bug once you add a promo code.
The promo code is displayed from a condition in tpl file '{if $cart.vouchers.added}'.

Reply:
Well then you shouldn't use `setInterval` this forces an useless overload every 100ms you perform this check just in case the promo code has been added

If this code is supposed to be executed when a promo code is added, I suppose it is added via javascript then you have to display your buttons correctly at this moment
If it is displayed thanks to smarty in a tpl file then it should already be on your page no? Unless it is updated via ajax?